cryptoid/src/adapter.ts

```diff
@@ -0,0 +1,47 @@
+import { Requester, Validator } from '@chainlink/external-adapter'
+import { Config, ExecuteWithConfig, ExecuteFactory } from '@chainlink/types'
+import { makeConfig, DEFAULT_ENDPOINT } from './config'
+import { util } from '@chainlink/ea-bootstrap'
+
+const customParams = {
+  blockchain: ['blockchain', 'coin'],
+  endpoint: false,
+}
+
+const endpointToApiParam: { [key: string]: string } = {
+  difficulty: 'getdifficulty',
+  height: 'getblockcount',
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, config) => {
+  const validator = new Validator(request, customParams)
+  if (validator.error) throw validator.error
+
+  Requester.logConfig(config)
+
+  const jobRunID = validator.validated.id
+  const endpoint = validator.validated.data.endpoint || DEFAULT_ENDPOINT
+  const blockchain = validator.validated.data.blockchain.toLowerCase()
+
+  const key = util.getRandomRequiredEnv('API_KEY')
```

Comment:
If you need some configuration in your adapter, it should be constructed in the `makeConfig` function and injected here to be used. You already have an API key setup in your config here:

```suggestion
  const key = config.apiKey
```